src/DependencyManagement/ManagedDependenciesPathDetector.cs

```diff
@@ -30,6 +31,11 @@ internal static class ManagedDependenciesPathDetector
         /// </summary>
         public static string GetManagedDependenciesPath(string functionAppRootPath)
         {
+            if (IsLinuxConsumptionOnLegion())
```

Comment:
Consider moving this check to a more intuitively obvious place, such as: https://github.com/Azure/azure-functions-powershell-worker/blob/25260ecd0b3f5fdb1e3a9297b7b80ec26245f80c/src/DependencyManagement/DependencyManager.cs#L107

This error should happen _only_ when the `requirements.psd1` contains any module references. The error should _not_ happen when `requirements.psd1` is absent, empty, or contains no module references. For example, a brand new app generated from a template should _not_ report this error. Chances are `GetManagedDependenciesPath` invocation does  not depend on the same factors. I suggest performing this check in `DependencyManager.Initialize`, just after we made sure that `_dependenciesFromManifest` is not empty.

Reply:
Thank you @AnatoliB for the suggestion. I have updated the code accordingly.